app/views/admin/AdministerUserModal.js

```diff
@@ -234,6 +235,42 @@ module.exports = (AdministerUserModal = (function () {
       return this.user.patch(options)
     }
 
+    onClickAddCreditsButton () {
+      const attrs = forms.formToObject(this.$('#credit-form'))
+      attrs.addCredits = parseInt(attrs.addCredits)
+      if (!_.all(_.values(attrs))) { return }
+      if (!(attrs.addCredits > 0)) { return }
+      if (!attrs.endDate) { return }
+      if (!attrs.creditType) { return }
+      attrs.endDate = attrs.endDate + ' ' + '23:59'
+      attrs.endDate = momentTimezone.tz(attrs.endDate, this.timeZone).toISOString()
+
+      this.state = 'creating-credits'
+      this.renderSelectors('#credit-form')
+
+      fetchJson('/db/user-credits/extra-credits', {
```

Comment:
move this fetchJson part to `app/core/api/user-credits` 
You might need to that userCredit to this file: https://github.com/codecombat/codecombat/blob/master/app/core/api/index.js and use it here